core/deployment/src/main/java/io/quarkus/deployment/steps/BannerProcessor.java

```diff
@@ -8,27 +8,30 @@
 
 import org.jboss.logging.Logger;
 
-import io.quarkus.banner.runtime.BannerRecorder;
+import io.quarkus.BannerConfig;
+import io.quarkus.IsBanner;
 import io.quarkus.builder.Version;
+import io.quarkus.deployment.IsDevelopment;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.ConsoleFormatterBannerBuildItem;
 import io.quarkus.deployment.builditem.HotDeploymentWatchedFileBuildItem;
 import io.quarkus.deployment.util.FileUtil;
+import io.quarkus.runtime.BannerRecorder;
 
 public class BannerProcessor {
 
     private static final Logger logger = Logger.getLogger(BannerProcessor.class);
 
-    @BuildStep(loadsApplicationClasses = true)
+    @BuildStep(loadsApplicationClasses = true, onlyIf = { IsBanner.class, IsDevelopment.class })
     @Record(ExecutionTime.RUNTIME_INIT)
     public ConsoleFormatterBannerBuildItem recordBanner(BannerRecorder recorder, BannerConfig config) {
         String bannerText = readBannerFile(config);
         return new ConsoleFormatterBannerBuildItem(recorder.provideBannerSupplier(bannerText));
     }
 
-    @BuildStep
+    @BuildStep(onlyIf = IsDevelopment.class)
```

Comment:
Shouldn't it be something like: 

```suggestion
    @BuildStep(onlyIf = { IsBanner.class, IsDevelopment.class })
```